### PR TITLE
Change kubeconfig context name from default to cluster ID

### LINKED ISSUE #5745: Change kubeconfig context name from "default" to cluster ID
**User story**

I'm a KKP user, managing more than one Kubernetes clusters. From my perspective, there are two main issues with having "default" context name set in user cluster's kubeconfig file:
- it's not easily possible to merge several such kubeconfig files into one for easy context switching (e.g. with [kubectx](https://github.com/ahmetb/kubectx) tool) because all share the same context name
-  it's useless to show the current Kubernetes context in the command line prompt, as it will just say "default" for all of them

**Acceptance criteria**

Change name of the context from "default" to cluster ID for user cluster kubeconfigs.

### PR DESCRIPTION
**Pull request by ahmedwaleedmalik**
**What this PR does / why we need it**:
Assign a unique identifier i.e. cluster ID for the `context` specified in the OIDC kubeconfig. Having it set as `default` makes it difficult to differentiate and switch context between multiple clusters.

**Which issue(s) this PR fixes**:

Fixes https://github.com/kubermatic/dashboard/issues/5745

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
The context name for admin Kubeconfig has been changed to the cluster ID from `default`.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/resources/kubeconfig.go [hunk 1]
```diff
@@ -185,9 +185,9 @@ func GetBaseKubeconfig(caCert *x509.Certificate, server, clusterName string) *cl
 				Server:                   server,
 			},
 		},
-		CurrentContext: KubeconfigDefaultContextKey,
+		CurrentContext: clusterName,
 		Contexts: map[string]*clientcmdapi.Context{
-			KubeconfigDefaultContextKey: {
+			clusterName: {
 				Cluster:  clusterName,
 				AuthInfo: KubeconfigDefaultContextKey,
```
Comment (xrstf) [inline pkg/resources/kubeconfig.go]: It feels kind of weird to me that we rename the context itself, but continue to use a constant named "KubeconfigDefaultContextKey" for what is not actually the context, but the AuthInfo. Can we rename the constant accordingly to like `KubeconfigDefaultAuthInfoKey` and adjust its doc comment?

Comment (xrstf) [reply pkg/resources/kubeconfig.go]: To be clear, renaming the context itself is not weird. :D

Comment (ahmedwaleedmalik) [reply pkg/resources/kubeconfig.go]: Done :D

**Review by xrstf — APPROVED** [review]
/approve

**Comment by ahmedwaleedmalik** [comment]
/hold
Unrelated tests failing due to vCenter being inaccessible.

**Pull request merged**